app/views/courses/CoursesView.coffee

```diff
@@ -81,6 +85,13 @@ module.exports = class CoursesView extends RootView
     if hocCourseInstance
       @courseInstances.remove(hocCourseInstance)
 
+    _.forEach Object.keys(courseIDs), (courseID) =>
+      levels = new Levels()
```

Comment:
A functional approach to building the list of courseIDs I think would be cleaner.

```
courseIDs = _.unique(courseInstances.pluck('courseID'))
```

Or, longer but a bit more readable, using lodash wrapper:

```
_(@courseInstances.models)
  .map((ci) -> ci.get('courseID'))
  .unique()
  .value()
```